juju/api.go

```diff
@@ -156,10 +156,11 @@ func NewAPIConnection(args NewAPIConnectionParams) (_ api.Connection, err error)
 				accountDetails.LastKnownAccess = st.ControllerAccess()
 			}
 		}
-		if ok && !user.IsLocal() && apiInfo.Tag == nil {
+		if ok && !user.IsLocal() && apiInfo.Tag == nil && accountDetails.Type != jujuclient.OAuth2DeviceFlowAccountDetailsType {
```

Comment:
On this line, accountDetails can be nil leading to a null pointer dereference. This is possible if the Juju CLI is running without a store.

Reply:
fixed